scipy/stats/_stats_py.py

```diff
@@ -6758,11 +6743,9 @@ def ttest_ind(a, b, *, axis=0, equal_var=True, nan_policy='propagate',
     """
     xp = array_namespace(a, b)
 
-    default_float = xp.asarray(1.).dtype
-    if xp.isdtype(a.dtype, 'integral'):
-        a = xp.astype(a, default_float)
-    if xp.isdtype(b.dtype, 'integral'):
-        b = xp.astype(b, default_float)
+    dtype = xp_result_type(a, b, force_floating=True, xp=xp)
+    a = xp.asarray(a, dtype=dtype)
+    b = xp.asarray(b, dtype=dtype)
```

Comment:
FWIW this looks cleaner and more readable to me than
```python
a, b = xp_broadcast_promote(a, b, force_floating=True, broadcast=False, xp=xp)
```
which makes a case for deleting the broadcast=True option from the signature of `xp_broadcast_promote`.

Reply:
Huh I think that's even why I added the `broadcast` argument, but I forgot to use it.

I drill think it's convenient to have a function that can *just* promote a bunch of arga.

How about calling it `xp_promote` leaving the `broadcast` argument with a default of `False` so `broadcast=True` is always explicit (but not used here).

Reply:
sounds good to me

Reply:
Alternatively, why not just skip the `broadcast=False` parameter and write `xp.broadcast_arrays(xp_promote(*arrays))` when needed?